scene/main/node.cpp

```diff
@@ -1838,12 +1838,19 @@ void Node::add_child(Node *p_child, bool p_force_readable_name, InternalMode p_i
 
 void Node::add_sibling(Node *p_sibling, bool p_force_readable_name) {
 	ERR_FAIL_COND_MSG(data.inside_tree && !Thread::is_main_thread(), "Adding a sibling to a node inside the SceneTree is only allowed from the main thread. Use call_deferred(\"add_sibling\",node).");
+	ERR_THREAD_GUARD
```

Comment:
This overlaps strongly with the line above, I don't think it should be added here

Reply:
I wan't sure so I just copied the commit that added those two lines together in `add_child`. If this was not intended I could remove it in this function too.

Reply:
`ERR_THREAD_GUARD` is just: `!is_accessible_from_caller_thread()`, which checks for:
```cpp
if (current_process_thread_group == nullptr) {
	// No thread processing.
	// Only accessible if node is outside the scene tree
	// or access will happen from a node-safe thread.
	return !data.inside_tree || is_current_thread_safe_for_nodes();
} else {
	// Thread processing.
	return current_process_thread_group == data.process_thread_group_owner;
}
```
So indeed it does matter in the specific edge case of `current_process_thread_group != nullptr`